compiler/rustc_ty_utils/src/abi.rs

```diff
@@ -702,30 +701,19 @@ fn fn_abi_adjust_for_abi<'tcx>(
                 // change their ABIs.
                 && abi != SpecAbi::RustIntrinsic
             {
-                match arg.layout.abi {
-                    // Handle similar to the way arguments with an `Abi::Aggregate` abi are handled
-                    // below, by returning arguments up to the size of a pointer (32 bits on x86)
-                    // cast to an appropriately sized integer.
-                    Abi::Scalar(s) if s.primitive() == Float(F32) => {
-                        // Same size as a pointer, return in a register.
-                        arg.cast_to(Reg::i32());
-                        return;
-                    }
-                    Abi::Scalar(s) if s.primitive() == Float(F64) => {
-                        // Larger than a pointer, return indirectly.
-                        arg.make_indirect();
-                        return;
-                    }
-                    Abi::ScalarPair(s1, s2)
-                        if matches!(s1.primitive(), Float(F32 | F64))
-                            || matches!(s2.primitive(), Float(F32 | F64)) =>
-                    {
+                let has_float = matches!(arg.layout.abi, Abi::Scalar(s) if matches!(s.primitive(), Float(_)))
+                    || matches!(arg.layout.abi, Abi::ScalarPair(s1, s2)
+                        if matches!(s1.primitive(), Float(_)) || matches!(s2.primitive(), Float(_)));
```

Comment:
I... don't find this more readable than the full
```rust
let has_float = match arg.layout.abi {
    pat if cond => true,
    _ => false,
};
```

Reply:
The thing that is throwing me off is the parens-counting combined with rustfmt's very... interesting choices on how to format this, btw, so I'd also be happy with anything that has slightly less parentheses.

Reply:
rustfmt didn't format this, it just skips the line. So I hand-formatted.

Reply:
I pushed a different way of writing this. Now rustfmt does format it and IMO not in a good way, as is often the case with `match`...

Reply:
Ahh, that would explain...

I don't _love_ this formatting but it's more tolerable to me, so thank you. I will try to deindent this code a bit in refactorings, then its formatting will matter less.

Reply:
e.g. I'd also accept this color for the bikeshed.
kinda can't explain why, tbh.
```rust
                let has_float = matches!(arg.layout.abi, Abi::Scalar(s)
                    if matches!(s.primitive(), Float(_)))
                    || matches!(arg.layout.abi, Abi::ScalarPair(s1, s2)
                        if matches!(s1.primitive(), Float(_))
                        || matches!(s2.primitive(), Float(_)));
```

Reply:
Uh I find that quite hard to read.^^

Reply:
yes, I thought so. 😅